src/utils/storage/index.ts

```diff
@@ -83,13 +85,15 @@ const StorageUtils = {
   otherProfileData: storageBuilder(StorageKeysEnum.OtherProfileData),
   lastPromptNotification: storageBuilder(StorageKeysEnum.LastPromptNotification),
   lastSelectedMenu: storageBuilder(StorageKeysEnum.LastSelectedMenu),
+  blockingStatus: storageBuilder(StorageKeysEnum.BlockingStatus),
+  blockedStatus: storageBuilder(StorageKeysEnum.BlockedStatus),
   clearAll: () => MMKVStorage.clearAll()
 };
 
 const clearAll = () => {
   const onboardingPassword = StorageUtils.onboardingPassword.get();
   MMKVStorage.clearAll();
-  StorageUtils.onboardingPassword.set(onboardingPassword);
+  StorageUtils.onboardingPassword.set(onboardingPassword || '');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `clearAll` function, you're setting the `onboardingPassword` to an empty string if it's undefined after clearing all storage. This could potentially lead to unexpected behavior if other parts of the application expect `onboardingPassword` to be `undefined` when it's not set. Consider leaving `onboardingPassword` as `undefined` if it was `undefined` before clearing the storage.
```diff
- StorageUtils.onboardingPassword.set(onboardingPassword || '');
+ if (onboardingPassword !== undefined) {
+   StorageUtils.onboardingPassword.set(onboardingPassword);
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->